mayastor/src/bdev/nexus/nexus_persistence.rs

```diff
@@ -87,6 +92,22 @@ impl Nexus {
                     }
                 });
             }
+            // Only update the state of the child if the precondition holds.
+            PersistOp::UpdateCond((uri, state, f)) => {
+                // Do not persist the state if predicate fails.
+                if !f(&nexus_info) {
+                    return;
+                }
+
+                let uuid =
+                    NexusChild::uuid(&uri).expect("Failed to get child UUID.");
+
+                nexus_info.children.iter_mut().for_each(|c| {
+                    if c.uuid == uuid {
+                        c.healthy = Self::child_healthy(&state);
+                    }
+                });
```

Comment:
rather than duplicating, would it be possible to make use of the same code above (used in the Update)? Though we can't simply call self.persist again because we've taken the lock.

Reply:
Yes, we can't use reentrance here as we're holding the lock.